src/js/client/editprefset-genericSettingsTable-widget.js

```diff
@@ -0,0 +1,206 @@
+    bindings: {
+    },
+    selectors: {
+        commonTermRow: ".pmt-commonterm-row",
+        prefsFilter: "#pmt-filter-container",
+        valueDisplayCell: ".pmt-value-display",
+
+        // Generic Prefs Filters
+        settingsSearchInput: "#pmt-settings-search-input",
+        addContextButton: ".pmt-add-context-button",
+
+        // mineAllWidget
+        mineAllSwitchContainer: "#pmt-mineAllSwitch-container",
+
+        // PSP items
+        pspShowCheckboxes: ".pmt-psp-show-checkbox",
+        pspMemoryCheckboxes: ".pmt-psp-memory-checkbox"
+    },
+    components: {
+        mineAllSwitch: {
+            type: "gpii.devpmt.mineAllSwitch.default",
+            createOnEvent: "onMarkupRendered",
+            container: "{that}.dom.mineAllSwitchContainer",
+            options: {
+                selectors: {
+                    initial: "#pmt-mineAllSwitch-area"
+                },
+                model: {
+                    settingsFilter: "{genericSettingsTableWidget}.model.settingsFilter",
+                    allSettingsCount: "{genericSettingsTableWidget}.model.commonTermUsageCounts.all",
+                    mySettingsCount: "{genericSettingsTableWidget}.model.commonTermUsageCounts.npset"
+                }
+            }
+        },
+        filter: {
+            type: "gpii.devpmt.filterWidget",
+            createOnEvent: "onMarkupRendered",
+            container: "{that}.dom.prefsFilter",
+            options: {
+                selectors: {
+                    initial: "#pmt-common-filter-area"
+                },
+                model: {
+                    filterText: "{genericSettingsTableWidget}.model.settingsSearch"
+                }
+            }
+        }
+    },
+    markupEventBindings: {
+        valueDisplayCell: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.editValueEvent"
+        },
+        addContextButton: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.events.openAddContextDialog.fire"
+        },
+        pspShowCheckboxes: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.editPspShow"
+        },
+        pspMemoryCheckboxes: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.editPspMemory"
+        }
+    },
+    listeners: {
+        "onCreate": [
+        ],
+        "onMarkupRendered": [
+            {
+                func: "{that}.updateSettingsFilter"
+            }
+        ]
+    },
+    modelListeners: {
+        "settingsSearch": {
+            func: "{that}.reRender",
+            excludeSource: ["init"]
+        },
+        "settingsFilter": {
+            func: "{that}.reRender",
+            excludeSource: ["init"]
+        },
+        "flatPrefs": {
+            func: "{that}.reRender",
+            excludeSource: ["init"]
+        }
+    }
+});
+
+gpii.devpmt.searchSettings = function (that, searchInput) {
+    that.applier.change("settingsSearch", searchInput);
```

Comment:
As above, use model listener to link up "settingsSearch" model path with "{that}.reRender" if it makes sense.

Reply:
removed, as there is already a modelListener for it.